app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,22 @@
+module SessionsHelper
+  # Logs in the given user.
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  # Returns the current logged-in user (if any).
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  # Returns true if the user is logged in, false otherwise.
+  def logged_in?
+    !current_user.nil?
```

Comment:
phân biệt present? và nil? giúp chị nhé

Reply:
dạ nil? dùng để check nil của object, nếu nil trả về true, ngược lại thì false. present? dùng để check not nil, not false, not empty của object, trả về true object đó not nil, not false, not empty